tools/ci/website_build.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -x
+
+neutral_status=78
+source_revision=$(git rev-parse HEAD)
+
+function modifies_relevant_files {
+  base_revision=$(
+    cat ${GITHUB_EVENT_PATH} | \
+      python -c 'import json, sys; print json.load(sys.stdin)["before"]'
+  )
+
+  git diff --name-only ${base_revision} | \
+    grep -E --silent '^(docs|tools)/'
+}
+
+if ! modifies_relevant_files ; then
+  echo No files related to the website have been modified. Exiting without
+  echo building.
+
+  exit ${neutral_status}
+fi
+
+git config --global user.email "wpt-pr-bot@users.noreply.github.com"
+git config --global user.name "wpt-pr-bot"
+
+cd docs
+
+pip install -r requirements.txt
+
+make html
+
+cd _build/html
+
+git init
+
+# Configure DNS
+echo web-platform-tests.org > CNAME
+
+# Disable Jekyll
+# https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/
+touch .nojekyll
+
+git add .
+
+cat <<HERE |
```

Comment:
We have now reached the kind of shell magic that leaves me guessing at what's going on :)

Reply:
This syntax was foreign to me, as well. A multi-line string should do just fine.